metadata-io/src/main/java/com/linkedin/metadata/graph/SiblingGraphService.java

```diff
@@ -92,11 +94,14 @@ private EntityLineageResult filterLineageResultFromSiblings(
       EntityLineageResult entityLineageResult,
       EntityLineageResult existingResult
   ) {
-    // 1) remove the source entities siblings from this entity's downstreams
+    Set<Urn> existingResultUrns = existingResult != null ?
+        existingResult.getRelationships().stream().map(LineageRelationship::getEntity).collect(Collectors.toSet()) : new HashSet<>();
+
+    // 1) remove duplicates and the source entities siblings from this entity's downstreams
     List<LineageRelationship> filteredRelationships = entityLineageResult.getRelationships()
         .stream()
-        .filter(lineageRelationship -> !allSiblingsInGroup.contains(lineageRelationship.getEntity())
-            || lineageRelationship.getEntity().equals(urn))
+        .filter(lineageRelationship -> !existingResultUrns.contains(lineageRelationship.getEntity()) &&
```

Comment:
I'll admit it's not immediately obvious to me, how does this remove duplicates? To do that wouldn't you want to convert to a set and then back to a list?

Reply:
ah so this doesn't remove duplicates from within `entityLineageResults` but filters out relationships that are already in `existingResultUrns` (so removing duplicates once we combine them).

But since this isn't very obvious and probably isn't the "Java" way - I can convert the `combinedResults` below into a set and then convert that to a list - going to try that and push it up!